tokio/src/fs/file.rs

```diff
@@ -753,20 +808,90 @@ impl AsyncWrite for File {
                     let n = buf.copy_from(src, me.max_buf_size);
                     let std = me.std.clone();
 
-                    let blocking_task_join_handle = spawn_mandatory_blocking(move || {
-                        let res = if let Some(seek) = seek {
-                            (&*std).seek(seek).and_then(|_| buf.write_to(&mut &*std))
-                        } else {
-                            buf.write_to(&mut &*std)
-                        };
+                    #[allow(unused_mut)]
+                    let mut data = Some((std, buf));
+
+                    let mut task_join_handle = None;
+
+                    #[cfg(all(
+                        tokio_unstable,
+                        feature = "io-uring",
+                        feature = "rt",
+                        feature = "fs",
+                        target_os = "linux"
+                    ))]
+                    {
+                        use crate::runtime::Handle;
+
+                        // Handle not present in some tests?
+                        if let Ok(handle) = Handle::try_current() {
+                            if handle.inner.driver().io().check_and_init()? {
+                                task_join_handle = {
+                                    use crate::{io::uring::utils::ArcFd, runtime::driver::op::Op};
+
```

Comment:
This is way too many levels of indentation. We need to pull some logic into a separate function.